cupy/_math/misc.py

```diff
@@ -239,6 +239,36 @@ def clip(a, a_min=None, a_max=None, out=None):
     ''')
 
 
+_float_heaviside_head = '''
+if (isnan(in0)) {
+    out0 = in0;
+}
+'''
+_heaviside = '''
+if (in0 == 0) {
+    out0 = in1;
+} else {
+    out0 = (in0 > 0);
+}
+'''
+_float_heaviside = _float_heaviside_head + " else " + _heaviside
+heaviside = _core.create_ufunc(
+    'cupy_heaviside',
+    ('bb->b', 'BB->B', 'hh->h', 'HH->H', 'ii->i', 'II->I',
+     'll->l', 'LL->L', 'qq->q', 'QQ->Q',
```

Comment:
numpy.heaviside always return floating values. Could you remove these combinations?
```py
>>> numpy.heaviside(0, 1)
1.0
```

Reply:
Fixed in 99825e52b7dd86e1da31664a202c13a93803242f.